pkg/http-handler/http-handler.go

```diff
@@ -42,14 +42,21 @@ type appRepositoryResponse struct {
 	AppRepository v1alpha1.AppRepository `json:"appRepository"`
 }
 
+func JSONError(w http.ResponseWriter, err interface{}, code int) {
+	w.Header().Set("Content-Type", "application/json; charset=utf-8")
+	w.Header().Set("X-Content-Type-Options", "nosniff")
+	w.WriteHeader(code)
+	json.NewEncoder(w).Encode(err)
+}
+
 func returnK8sError(err error, w http.ResponseWriter) {
 	if statusErr, ok := err.(*k8sErrors.StatusError); ok {
 		status := statusErr.ErrStatus
 		log.Infof("unable to create app repo: %v", status.Reason)
-		http.Error(w, status.Message, int(status.Code))
+		JSONError(w, statusErr.ErrStatus, int(status.Code))
 	} else {
 		log.Errorf("unable to create app repo: %v", err)
-		http.Error(w, err.Error(), http.StatusInternalServerError)
+		JSONError(w, err.Error(), http.StatusInternalServerError)
 	}
 }
 
```

Comment:
there are a few places in which we are just returning the plain error (without the JSON wrapper). I guess we should standardize with this way of returning an error. I am talking about the lines: 72, 130 and 144

Reply:
Thanks, replace all three cases of `http.Error` with `JSONError`.